File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int low = 0;
        int high = num;
        while(low<=high){
            long mid = low + (high-low)/2;

            long target = mid * mid;
            if(target < num){
                low = mid + 1;
            }
            else if (target == num){
                return true;
            }
            else{
                high = mid -1;
            }
        }
        return false;
        
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        low = 0
        high = num
        while(low<=high):
            mid = low + (high-low)//2
            target = mid * mid
            if(target==num):
                return True

            elif(target< num):
                low = mid+1

            else:
                high = mid - 1

        return False